spec/acceptance/suites/upgrade/00_upgrade_spec.rb

```diff
@@ -16,64 +18,147 @@
       on puppetserver, 'mkdir -p /var/isos'
-      iso_files.each do |iso_file|
-        puppetserver.do_rsync_to iso_file, '/var/isos/'
+      iso_files.each do |file|
+        puppetserver.do_rsync_to file, "/var/isos/#{File.basename(file)}"
       end
     end
 
-    #  Upgrade process derived from:
-    #
-    #    https://github.com/simp/simp-doc/blob/8277eab0a38f3c995a762e017fe0cc65c7ac3bb8/docs/getting_started_guide/ISO_Install/SIMP_Server_Installation.rst
-    #
     it 'runs the unpack_dvd script' do
-      iso_files.each do |iso_file|
-        on puppetserver, "unpack_dvd /var/isos/#{iso_file}"
+      upload_rpms_to_yum_repo # inject rpms
+      iso_files.each do |file|
+        on puppetserver, "unpack_dvd /var/isos/#{File.basename(file)}"
       end
       on puppetserver, 'yum clean all; yum makecache'
-      on puppetserver, 'puppet agent -t', :acceptable_exit_codes => [2]
+    end
+
+    it 'runs `yum update`' do
+      errata_for_simp5383__yum_excludes(:add)
+      on puppetserver, 'yum --rpmverbosity=warn -y update'
+    end
+
+    it 'runs `puppet agent -t` to apply changes' do
+      on puppetserver, "#{puppet_agent_t}.1", :acceptable_exit_codes => [2]
+
+      if errata_for_simp5383__yum_excludes(:remove)
+        # SIMP-5383: run agent an extra time to upgrade the agent
+        on puppetserver, "#{puppet_agent_t}.2", :acceptable_exit_codes => [2]
+      end
+    end
+
+    it 'runs `puppet agent -t` idempotently' do
+      on puppetserver, "#{puppet_agent_t}.3", :acceptable_exit_codes => [0]
+    end
+
+    # Helper methods
+    # --------------------------------------------------------------------------
+
+    # based on the env var `BEAKER_upgrade__new_simp_iso_path` or a file glob,
+    # returns an Array of isos if found, or fails if not
+    def local_iso_files_matching(default_file_glob)
+      isos = if (iso_files = ENV['BEAKER_upgrade__new_simp_iso_path'])
+               iso_files.to_s.split(%r{[:,]})
+             else
+               Dir[default_file_glob]
+             end
+      return isos unless isos.empty?
+      raise <<-NO_ISO_FILE_ERROR.gsub(%r{^ {6}}, '')
+
+        --------------------------------------------------------------------------------
+        ERROR: No SIMP ISO(s) to upload for upgrade!
+        --------------------------------------------------------------------------------
+
+        This test requires at least one newer SIMP .iso
+
+        You can provide .iso files either by setting the environment variable:
+
+            BEAKER_upgrade__new_simp_iso_path=/path/to/iso-file.iso
+
+        Or:
+
+        Place a file that matches the glob '#{default_file_glob}'
+        into the top directory of this project.
+
+        --------------------------------------------------------------------------------
+
+      NO_ISO_FILE_ERROR
+    end
+
+    # Upload matching local RPMs into the puppetserver's yum repo
+    def upload_rpms_to_yum_repo(base_dir = '.', rpm_globs = ['*.noarch.rpm'])
```

Comment:
Why not '*.rpm' ?  I don't see any reason to restrict to noarch RPMs.

Reply:
I laid in some groundwork to split out where to pick up RPMs based on major os version/arch/ etc, but it wasn't a priority for this sprint so I stuck with the safest option.

Reply:
Well, since they are going in x86_64, seems like *noarch.rpm and *x86_64.rpm would both be safe